server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -215,24 +215,31 @@ class InteractiveSession(
     stop()
   }
 
-  def executeStatement(content: ExecuteRequest): Statement = {
+  def executeStatement(content: ExecuteRequest): Statement = synchronized {
```

Comment:
I'm a change for session recovery that slashes with this change a lot.
https://github.com/hdinsight/livy/commit/6c7f15b58d0fd6844021930b6d6087aaee8b0a9a